pkg/controller/networkpolicy/clustergroup.go

```diff
@@ -108,11 +108,17 @@ func (n *NetworkPolicyController) deleteClusterGroup(oldObj interface{}) {
 		}
 	}
 	key := internalGroupKeyFunc(og)
-	klog.V(2).Infof("Deleting internal Group %s", key)
-	err := n.internalGroupStore.Delete(key)
-	if err != nil {
-		klog.Errorf("Unable to delete internal Group %s from store: %v", key, err)
+	grpObj, found, _ := n.internalGroupStore.Get(key)
+	if found {
+		grp := grpObj.(*antreatypes.Group)
+		klog.V(2).Infof("Deleting internal Group %s", key)
+		err := n.internalGroupStore.Delete(key)
+		if err != nil {
+			klog.Errorf("Unable to delete internal Group %s from store: %v", key, err)
+		}
+		n.triggerParentGroupSync(grp)
```

Comment:
we have been triggering CNP updates from workers.. is it possible to do so in syncInternalGroup?

Reply:
It will be very tricky to do so in `syncInternalGroup`, in case of deletion. If a group key is processed by `syncInternalGroup`, the controller would expect the internal group still exist in the internal group store, otherwise it won't do anything. So in this case, if we want to rely on `syncInternalGroup` for parent group syncs, we would need to delete the internal group from the internal store after `syncInternalGroup`. However, at the time of parent group sync we also want to ensure that the childGroup is deleted from internal store, so that we don't include outdated members. Since this process is async, it's hard for us to ensure that `syncInternalGroup` of the parent happens after the internal group deletion.

Reply:
hmm im worried this comes in the path of the synchronous Delete handlers and may cause perf issue. maybe add a TODO comment here to explain why its done here now and perhaps we should look in to moving this to the workers. let me also dig deeper if we can have an alternative here..

Reply:
you plan to add a TODO here?

Reply:
Thanks for the reminder. Added TODO.